src/test/java/edu/ucsb/cs56/ucsb_open_lab_scheduler/end2end/AdminPageEnd2EndTest.java

```diff
@@ -0,0 +1,109 @@
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.junit.*;
+import org.junit.runner.RunWith;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+
+/**
+ * This test runs a complete end-to-end test on the project as a single test.
+ * Starts by logging in as a student, submitting a review, and reviewing 5 ideas
+ *
+ * If you are trying to duplicate this test in a project, make sure to copy the
+ * html template under the test/resources/__files/ directory!
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT, properties="spring.datasource.name=end2endtest")
+public class AdminPageEnd2EndTest {
+
+    static {
+        System.setProperty("spring.security.oauth2.client.provider.wiremock.authorization-uri", "http://localhost:8077/oauth/authorize");
+        System.setProperty("spring.security.oauth2.client.provider.wiremock.token-uri", "http://localhost:8077/oauth/token");
+        System.setProperty("spring.security.oauth2.client.provider.wiremock.user-info-uri", "http://localhost:8077/userinfo");
+        System.setProperty("spring.security.oauth2.client.provider.wiremock.user-name-attribute", "sub");
+        System.setProperty("spring.security.oauth2.client.registration.wiremock.provider", "wiremock");
+        System.setProperty("spring.security.oauth2.client.registration.wiremock.authorization-grant-type", "authorization_code");
+        System.setProperty("spring.security.oauth2.client.registration.wiremock.scope", "email");
+        System.setProperty("spring.security.oauth2.client.registration.wiremock.redirect-uri", "http://localhost:8080/login/oauth2/code/{registrationId}");
+        System.setProperty("spring.security.oauth2.client.registration.wiremock.clientId", "wm");
+        System.setProperty("spring.security.oauth2.client.registration.wiremock.clientSecret", "whatever");
+    }
+
+    private WebDriver webDriver;
+
+
+    @Rule
+    public WireMockRule mockOAuth2Provider = new WireMockRule(wireMockConfig()
+            .port(8077)
+            .extensions(new ResponseTemplateTransformer(true)));
+
+    @Before
+    public void setUp() {
+        // Setup ChromeDriver (aided by the WebDriverManager)
+        WebDriverManager.chromedriver().setup();
+        ChromeOptions options = new ChromeOptions();
+        //Run browser in headless mode for CI
+        options.addArguments("--headless");
+        webDriver = new ChromeDriver(options);
+
+        // Configure mock oauth endpoints
+        mockOAuth2Provider.stubFor(get(urlPathEqualTo("/favicon.ico")).willReturn(notFound()));
```

Comment:
This might also be a good candidate to be factored out into some kind of test helper...